subiquity/server/controllers/install.py

```diff
@@ -72,6 +73,18 @@ def feed(self, line):
             self.traceback.append(line)
 
 
+@attr.s(auto_attribs=True)
+class CurtinInstallStep:
+    """ Represents the parameters of a single step (i.e., invocation of curtin
+    install). """
+    name: str
+    stages: List[str]
+    config_file: Path
+    log_file: Path
+    error_file: Path
+    acquire_config: Callable[["CurtinInstallStep", Path], Dict[str, Any]]
```

Comment:
Having a callable to get the config feels a bit complicated. But I guess that can be made use of or cleaned up as we continue this work.